src/instructlab/model/train.py

```diff
@@ -520,20 +585,328 @@ def get_files(directory: str, pattern: str) -> list[str]:
+
+    os.makedirs(ckpt_path)
+    os.makedirs(eval_cache_path)
+
+
+def _training_phase(
+    train_args: TrainingArgs,
+    torch_args: TorchrunArgs,
+    data_path: str,
+    model_override: typing.Optional[str] = None,
+    ckpt_dir: typing.Optional[str] = None,
+) -> None:
+    """A single step of end-to-end training."""
+
+    # Third Party
+    from instructlab.training import run_training  # pylint: disable=no-name-in-module
+
+    # override whatever got injected at a higher level
+    logger.debug(
+        f"Phased Training -- training phase -- Overriding data_path: {train_args.data_path} with {data_path}"
+    )
+    train_args.data_path = data_path
+
+    if ckpt_dir:
+        train_args.ckpt_output_dir = ckpt_dir
+
+    if model_override:
+        logger.debug(
+            f"Phased Training -- training phase -- Overriding model_path: {train_args.model_path} with {model_override}"
+        )
+        train_args.model_path = model_override
+
+    logger.debug(
+        f"Phased Training -- training phase -- Current training args for phase: {pp.pformat(train_args)}"
+    )
+
+    try:
+        run_training(train_args=train_args, torch_args=torch_args)
+    except Exception as e:
+        logging.fatal(e)
+        sys.exit(1)
+
+
+def _mmlu(ckpt_path: str) -> float:
+    # Third Party
+    from instructlab.eval.mmlu import MMLU_TASKS, MMLUEvaluator
+    import torch
+
+    evaluator = MMLUEvaluator(ckpt_path, tasks=MMLU_TASKS, few_shots=5)
+
+    mmlu_results: tuple[float, dict] = evaluator.run()
+    ckpt_score: float = mmlu_results[0]
+
+    logging.debug("Phased Training -- MMLU eval phase -- Clearing PyTorch cache")
+    torch.cuda.empty_cache()
+
+    return ckpt_score
+
+
+def _mtbench(ctx, ckpt_path: str, eval_cache_path: str, judge_name: str) -> float:
+    # TODO: run all generations in serial and then do all judgements at once to save time loading/unloading prometheus.
+
+    # dynamic import- eval code not used outside of this path.
+    # Third Party
+    from instructlab.eval.mt_bench import MTBenchEvaluator
+    import torch
+
+    # First Party
+    from instructlab.model.evaluate import launch_server
+
+    # hard-override for local insecure vLLM serving
+    ctx.params["tls_client_cert"] = None
+    ctx.params["tls_client_key"] = None
+    ctx.params["tls_client_passwd"] = None
+    ctx.params["tls_insecure"] = True
+
+    evaluator = MTBenchEvaluator(
+        model_name=ckpt_path,
+        judge_model_name=judge_name,
+        output_dir=eval_cache_path,
+        merge_system_user_message=True,
+    )
+
+    server = None
+    model_serve_url = None
+    try:
+        logger.debug("Starting model server for mt-bench answer generation")
+        server, model_serve_url = launch_server(
+            ctx=ctx,
+            model=ckpt_path,
+            model_name=ckpt_path,
+            gpus=torch.cuda.device_count(),
+            max_workers=multiprocessing.cpu_count(),
```

Comment:
Should these two be configurable? I think eval supports that. If not now, please leave a TODO comment

Reply:
added a comment for this